fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraBaseResource.java

```diff
@@ -122,13 +121,30 @@ protected void setUpJMSInfo(final UriInfo uriInfo, final HttpHeaders headers) {
     /**
      * Produce a baseURL for JMS events using the system property fcrepo.jms.baseUrl of the form http[s]://host[:port],
      * if it exists.
-     *
+     * <p>
+     * Implementation note: forwards to {@link #getBaseUrlProperty(UriInfo)}, using the internal {@code UriInfo}.
+     * </p>
      * @return String the base Url
      */
     protected String getBaseUrlProperty() {
-        final String propBaseURL = System.getProperty("fcrepo.jms.baseUrl", "");
+        return getBaseUrlProperty(uriInfo);
+    }
+
+    /**
+     * Produce a baseURL for JMS events using the system property fcrepo.jms.baseUrl of the form http[s]://host[:port],
+     * if it exists.
+     *
+     * @param uriInfo used to build the base url
+     * @return String the base Url
+     */
+    protected String getBaseUrlProperty(final UriInfo uriInfo) {
```

Comment:
Happy to reduce the visibility and remove the other method, but that would also break any binary compat (e.g. any labs or exts that may have overridden the method).  If that's OK I'll make the updates!